### ISSUE TITLE
Update Wiki installation instructions for V4

### ISSUE THREAD

**Issue by ricardoV94** [pymc-devs/pymc#5603]
We could mention the PyPI beta release instead?

https://github.com/pymc-devs/pymc/wiki/Installation-Guide-(Linux)#pymc-v4-installation

**Comment by fonnesbeck** [comment]
I've updated instructions for Windows. There was some debate as to whether to recommend the Microsoft Visual C compilers versus mingw. They currently state that if you have VC/C++ on your system (or another compiler), great, if not then here's how to get mingw. All works great on my Windows system, but would be nice to have folks try to replicate.

**Comment by fonnesbeck** [comment]
Linux instructions updated [here](https://github.com/pymc-devs/pymc/wiki/Installation-Guide-(Linux)). Confirmed that it works on Debian and Ubuntu.

**Comment by ricardoV94** [comment]
This part on Windows guide looks wrong:

> Note that if you are updating to v4 over a previous version of PyMC, you may need to remove Theano and install Theano-PyMC before updating:
>```
>conda remove theano
>conda install -c conda-forge theano-pymc
>```

**Comment by ghost** [comment]
That's left over from the v3 instructions. I will fix it.

**Comment by twiecki** [comment]
@cfonnesbeck is this done?

**Comment by cluhmann** [comment]
The Mac instructions have been updated, but there is still cruft left from its previous iteration.  Do we still want to keep the "install from the development branch" and "install from a git clone" options?  The dev branch/git clone options point to pymc (i.e., v4).  And the Mac install instructions already have v4, v3 on M1, and v3 on Intel.  Plus theano-PyMC instructions.

**Comment by twiecki** [comment]
No, let's gear it towards installing 4.0.0 final only and push it out.

**Comment by cluhmann** [comment]
Sounds good.  I will update and then check back in here so someone can look it over (e.g., someone who actually owns a Mac).

**Comment by cluhmann** [comment]
Ok, that was actually easy.  Do we want a specific warning against installing via homebrew?  @fonnesbeck  found that there were some tricky steps:

>So, the Homebrew Python install is not easy. There does not seem to be universal NetCDF4 wheels yet (scheduled for v1.6) so there are a bunch of manual steps.

**Comment by cluhmann** [comment]
Two more things:

1 - The [linux installation instructions](https://github.com/pymc-devs/pymc/wiki/Installation-Guide-(Linux)) currently include instructions for installing jax/numpyro and blackjax.  Do we need/want those for Mac/Windows?  Do the instructions differ at all on those platforms?
2 - The linux instructions include pymc3 installation as a secondary (i.e., they are at the bottom.  Should we just eliminate those altogether?

**Comment by fonnesbeck** [comment]
1. We probably want them on the Mac side, but Jax for Windows is not really a thing yet.
2. The PyMC3 instructions should still be included somewhere, yes. Could have been a subsection of its own, but a single sentence was sufficient on Linux (at least on the system I tested it on).

**Comment by cluhmann** [comment]
Sure.  I can add the v3 instructions back into the Mac instructions (though there are 2 sets, one for M1 and one for Intel).

**Comment by cluhmann** [comment]
Ok.  I just copied the Jax/blackjax installation instructions from the linux page to the Mac page (not sure if installation works the same?).

Feel free to look over and close if everything looks ok.

**Comment by cluhmann** [comment]
Oh, and yes/no on warning against homebrew?

**Comment by ricardoV94** [comment]
The conda installation are a bit all over the place:
1. Linux does not mention the `conda-forge` channel and still has the `--pre` flag instruction. 
2. MacOS includes `python=3.9` in the conda instructions (why?)
3. Windows includes a bunch of other dependencies explicitly `libpython mkl-service numba python-graphviz scipy arviz` (why?) and still has the `--pre` flag instruction

Let's update the instructions as if V4 was already released and available in conda-forge?

**Comment by twiecki** [comment]
@ricardoV94 Yep, let's just fix those.

For windows it does require numba for some odd reason, but scipy and arviz shouldn't be required.

**Comment by cluhmann** [comment]
The `python=3.9` on Mac was there before we started revising things.  I removed it from the v4 instructions. Is any specific version of python needed for v3 on Mac, or can that go too?

**Comment by ricardoV94** [comment]
> The `python=3.9` on Mac was there before we started revising things. I removed it from the v4 instructions. Is any specific version of python needed for v3 on Mac, or can that go too?

There shouldn't be

**Comment by ricardoV94** [comment]
It seems like jax and numpyro are not available on conda unlike what the Wiki suggested (I changed it to pip), can someone confirm?

**Comment by ghost** [comment]
I would leave the `--pre` flag on for now until the actual release. Otherwise users will get v2 in the interim. It won't take long to remove it when needed.

**Comment by ricardoV94** [comment]
> I would leave the `--pre` flag on for now until the actual release. Otherwise users will get v2 in the interim. It won't take long to remove it when needed.

Okay, only MacOS does not mention it

**Comment by cluhmann** [comment]
>Okay, only MacOS does not mention it

That's because there are no pip instructions for v4 on the Mac (not for pymc anyway).

**Comment by ferrine** [comment]
Linux and Mac OS instructions duplicate each other, I move them under pymc docs page so we have everything on the documentation page

**Comment by twiecki** [comment]
@ferrine I think they are slightly different, because for Mac we also need to talk about not installing MKL.

**Comment by ferrine** [comment]
I've updated the docs to reflect that. Should we remove install instructions from wiki as well?

**Comment by twiecki** [comment]
@ferrine where did you change stuff?

**Comment by twiecki** [comment]
This is the PR https://github.com/pymc-devs/pymc/pull/5835

**Comment by twiecki** [comment]
They should be up-to-date now. Let's track moving them to the docs separately.
